src/js/pages/services/task-details/TaskDetail.js

```diff
@@ -0,0 +1,238 @@
+import mixin from 'reactjs-mixin';
+/* eslint-disable no-unused-vars */
+import React from 'react';
+/* eslint-enable no-unused-vars */
+import {RouteHandler} from 'react-router';
+import {StoreMixin} from 'mesosphere-shared-reactjs';
+
+import MarathonStore from '../../../stores/MarathonStore';
+import MesosStateStore from '../../../stores/MesosStateStore';
+import PageHeader from '../../../components/PageHeader';
+import RequestErrorMsg from '../../../components/RequestErrorMsg';
+import ServicesBreadcrumb from '../../../components/ServicesBreadcrumb';
+import TaskDirectoryStore from '../../../stores/TaskDirectoryStore';
+import TaskStates from '../../../constants/TaskStates';
+
+import InternalStorageMixin from '../../../mixins/InternalStorageMixin';
+import TabsMixin from '../../../mixins/TabsMixin';
+
+const SERVICES_TABS = {
+  'services-task-details-tab': 'Details',
+  'services-task-details-files': 'Files',
+  'services-task-details-logs': 'Logs'
+};
+
+const NODES_TABS = {
+  'nodes-task-details-tab': 'Details',
+  'nodes-task-details-files': 'Files',
+  'nodes-task-details-logs': 'Logs'
+};
+
+const METHODS_TO_BIND = [
+  'onTaskDirectoryStoreError',
+  'onTaskDirectoryStoreSuccess'
+];
+
+class TaskDetail extends mixin(InternalStorageMixin, TabsMixin, StoreMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.tabs_tabs = {};
+
+    this.state = {
+      directory: null,
+      expandClass: 'large',
+      selectedLogFile: null,
+      taskDirectoryErrorCount: 0
+    };
+
+    this.store_listeners = [
+      {name: 'state', events: ['success'], listenAlways: false},
+      {name: 'summary', events: ['success'], listenAlways: false},
+      {name: 'taskDirectory', events: ['error', 'success']}
+    ];
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentWillMount() {
+    super.componentWillMount(...arguments);
+    this.tabs_tabs = Object.assign({}, SERVICES_TABS);
+    if (this.props.params.nodeID != null) {
+      this.tabs_tabs = Object.assign({}, NODES_TABS);
+    }
+    this.updateCurrentTab();
+  }
+
+  updateCurrentTab() {
+    let routes = this.context.router.getCurrentRoutes();
+    let currentTab = routes[routes.length - 1].name;
+    if (currentTab != null) {
+      this.setState({currentTab});
+    }
+  }
+
+  componentDidMount() {
+    super.componentDidMount(...arguments);
+    this.store_removeEventListenerForStoreID('summary', 'success');
```

Comment:
I did not add this. This is not new code as you can see here: https://github.com/dcos/dcos-ui/compare/mlunoe/make-task-detail-tabs-routable-review...mlunoe/make-task-detail-tabs-routable?expand=1#diff-1daba2ecad3f911b9b723467f7d09e9bL64

However, I'm not exactly sure why this is how it is.